components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -357,8 +357,7 @@ void AdsServiceImpl::OnTabUpdated(
     return;
   }
 
-  const bool is_incognito = profile_->IsOffTheRecord() ||
-      brave::IsTorProfile(profile_);
+  const bool is_incognito = !brave::IsRegularProfile(profile_);
```

Comment:
nit: Hmm, this will work as expected but I think this could make some confusing because non incogito profile could be false from `brave::IsRegularProfile()`.

Reply:
I would like to do a minimum changes on original code here, maybe @tmancey could suggest what's preferred here and we could do a small quick follow-up on this?

Reply:
IMO, using `is_regular` seems more clear.
and calling  `bat_ads_->OnTabUpdated(tab_id.id(), url.spec(), is_active, is_browser_active, is_regular);`

Reply:
Actually I'm a bit confusing why bat_ads_ need to take is_incognito param at all, why do we expect this function to be executed if we don't expect to create ads service in the first place?

Reply:
Yeah, I also think we don't need to use `is_incognito` or replace it with DCHECK.

Reply:
Yeah, I think it might be better to separate this small piece out into a separate PR to keep this one simple and focused.

Reply:
Agreed this is legacy code which is no longer needed. We already have a `DCHECK`. I have raised https://github.com/brave/brave-browser/issues/11648. Thanks